app/views/orders/show.html.erb

```diff
@@ -8,17 +8,54 @@
   </div>
 <% end %>
 
+
 <div>Total price: <%= @order.total %></div>
 <%= form_tag charges_path do %>
+  <div>Name: <%= text_field_tag 'name', '', onblur: 'checkMandatoryValues()'%></div>
+  <div>Address 1: <%= text_field_tag 'address1', '', onblur: 'checkMandatoryValues()' %></div>
+  <div>Address 2: <%= text_field_tag 'address2' %></div>
+  <div>Postal Code: <%= text_field_tag 'postal_code', '', onblur: 'checkMandatoryValues()' %></div>
+  <div>City: <%= text_field_tag 'city', '', onblur: 'checkMandatoryValues()' %></div>
+  <div>E-mail: <%= text_field_tag 'email', '', onblur: 'checkMandatoryValues()' %></div>
+  <div>Phone: <%= text_field_tag 'phone_number', '', onkeyup: 'checkMandatoryValues()' %></div>
+
   <% if flash[:error].present? %>
     <div id="error_explanation">
       <p><%= flash[:error] %></p>
     </div>
   <% end %>
 
+  <div id="mandatoryAddressMessage">Customer Information Must be Filled in</div>
```

Comment:
`mandatoryAddressMessage` should be in other format  -> `mandatory-address-message`

Reply:
dangit! i'll get the hang of it eventually... fixing now